controllers/cluster_controller.go

```diff
@@ -120,15 +121,33 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		return ctrl.Result{}, err
 	}
 
-	if !cluster.ObjectMeta.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(ctx, cluster)
+	return ctrl.Result{}, nil
+}
+
+// reconcile handles cluster reconciliation.
+func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster) (err error) {
+	// If object doesn't have a finalizer, add one.
+	if !util.Contains(cluster.Finalizers, clusterv1.ClusterFinalizer) {
+		cluster.Finalizers = append(cluster.ObjectMeta.Finalizers, clusterv1.ClusterFinalizer)
 	}
 
-	return ctrl.Result{}, nil
+	// TODO(vincepri): These can be generalized with an interface and possibly a for loop.
+	errors := []error{}
+	errors = append(errors, r.reconcileInfrastructure(ctx, cluster))
+
+	// Determine the return error, giving precedence to the first non-nil and non-requeueAfter errors.
+	for _, e := range errors {
```

Comment:
Would it be better to aggregate the errors instead?

Reply:
We definitely can, and only aggregate the ones that aren't RequeueAfter

Reply:
Done